clang/lib/Driver/ToolChains/Darwin.cpp

```diff
@@ -3291,6 +3291,39 @@ void Darwin::addClangTargetOptions(
     if (!RequiresSubdirectorySearch)
       CC1Args.push_back("-fno-modulemap-allow-subdirectory-search");
   }
+
+  if (getTriple().isArm64e()) {
```

Comment:
There's already a block up in `MachO::addClangTargetOptions`.  But it's been committed separately from our upstreaming effort, so is definitely outdated; can you update it instead